apps/builder/app/builder/features/seo/project-redirects.tsx

```diff
@@ -0,0 +1,224 @@
+      */
+
+      if (newPath.startsWith("/") === true) {
+        if (existingPaths.has(newPath) === false) {
+          return ["This path doesn't exist in the project"];
+        }
+      }
+
+      return [];
+    }
+
+    return newPathValidationResult.error.format()._errors;
+  };
+
+  const handleOnKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === "Enter") {
+      handleAddRedirect();
+    }
+  };
+
+  const handleAddRedirect = () => {
+    const validOldPath = validateOldPath(oldPath);
+    const validNewPath = validateNewPath(newPath);
+
+    if (validOldPath.length > 0 || validNewPath.length > 0) {
+      return;
+    }
+
+    props.onSettingsChange({
+      ...props.settings,
+      redirects: [{ old: oldPath, new: newPath }, ...redirects],
+    });
+    setOldPath("");
+    setNewPath("");
+  };
+
+  const handleDeleteRedirect = (index: number) => {
+    const newRedirects = [...redirects];
+    newRedirects.splice(index, 1);
+    props.onSettingsChange({
+      ...props.settings,
+      redirects: newRedirects,
+    });
+  };
+
+  return (
+    <>
+      <Separator />
+      <Grid gap={2} css={{ mx: theme.spacing[5], px: theme.spacing[5] }}>
+        <Label sectionTitle>301 Redirects</Label>
+        <Text color="subtle">
+          Redirects old URLs to new ones so that you don’t lose any traffic or
+          search engine rankings.
+        </Text>
+
+        <Flex gap="2" align="center">
+          <InputErrorsTooltip
+            errors={oldPathErrors.length > 0 ? oldPathErrors : undefined}
+            side="top"
+            css={{ zIndex: theme.zIndices["1"] }}
+          >
+            <InputField
+              type="text"
+              placeholder="/old-path"
+              value={oldPath}
+              css={{ flexGrow: 1 }}
+              onKeyDown={handleOnKeyDown}
+              onChange={handleOldPathChange}
+              color={oldPathErrors.length === 0 ? undefined : "error"}
+            />
+          </InputErrorsTooltip>
+          <ArrowRightIcon />
+
+          <InputErrorsTooltip
+            errors={newPathErrors.length > 0 ? newPathErrors : undefined}
+            side="top"
+            css={{ zIndex: theme.zIndices["2"] }}
+          >
+            <InputField
+              type="text"
+              placeholder="/new-path or URL"
+              value={newPath}
+              onKeyDown={handleOnKeyDown}
+              onChange={handleNewPathChange}
+              color={newPathErrors.length === 0 ? undefined : "error"}
+            />
+          </InputErrorsTooltip>
+
+          <Button
+            disabled={isValidRedirects === false || oldPath === newPath}
+            onClick={handleAddRedirect}
+            css={{ flexShrink: 0 }}
+          >
+            Add
+          </Button>
+        </Flex>
+      </Grid>
+
+      {redirectKeys.length > 0 ? (
+        <Grid css={{ px: theme.spacing[5] }}>
+          <ScrollArea css={{ maxHeight: theme.spacing[22] }}>
```

Comment:
We don't need nested scroll
<img width="422" alt="image" src="https://github.com/webstudio-is/webstudio/assets/5635476/e59b81a6-7e0c-4156-bcbe-ea033d1bcfd7">

Reply:
I am assuming there could be dozens of redirects and this will be a problem if you want to get anything after that section, unless we keep this section always as a last one

With more sections added we probably are going to add a navigation on the left and each section will be just it's own page

Reply:
actually, ok lets kick out the inner scroll, and just keep that section as a last one for now